modules/svg/image_loader_svg.cpp

```diff
@@ -30,132 +30,120 @@
 
 #include "image_loader_svg.h"
 
-#include <nanosvg.h>
-#include <nanosvgrast.h>
-
-void SVGRasterizer::rasterize(NSVGimage *p_image, float p_tx, float p_ty, float p_scale, unsigned char *p_dst, int p_w, int p_h, int p_stride) {
-	nsvgRasterize(rasterizer, p_image, p_tx, p_ty, p_scale, p_dst, p_w, p_h, p_stride);
-}
-
-SVGRasterizer::SVGRasterizer() {
-	rasterizer = nsvgCreateRasterizer();
-}
-
-SVGRasterizer::~SVGRasterizer() {
-	nsvgDeleteRasterizer(rasterizer);
-}
-
-SVGRasterizer ImageLoaderSVG::rasterizer;
-
-inline void change_nsvg_paint_color(NSVGpaint *p_paint, const uint32_t p_old, const uint32_t p_new) {
-	if (p_paint->type == NSVG_PAINT_COLOR) {
-		if (p_paint->color << 8 == p_old << 8) {
-			p_paint->color = (p_paint->color & 0xFF000000) | (p_new & 0x00FFFFFF);
+#include "core/os/memory.h"
+#include "core/variant/variant.h"
+
+#include <thorvg.h>
+
+void ImageLoaderSVG::_replace_color_property(String p_prefix, String &p_string) {
+	// Replace colors in the SVG based on what is configured in `replace_colors`.
+	// Used to change the colors of editor icons based on the used theme.
+	String temp_color = "000000";
+	int idx = 0;
+	while (true) {
+		idx = p_string.find(p_prefix, idx);
+		if (idx == -1 || idx > p_string.length() - 9) {
+			break;
 		}
-	}
-
-	if (p_paint->type == NSVG_PAINT_LINEAR_GRADIENT || p_paint->type == NSVG_PAINT_RADIAL_GRADIENT) {
-		for (int stop_index = 0; stop_index < p_paint->gradient->nstops; stop_index++) {
-			if (p_paint->gradient->stops[stop_index].color << 8 == p_old << 8) {
-				p_paint->gradient->stops[stop_index].color = p_new;
-			}
+		idx += p_prefix.length();
+		if (p_string[idx] != '#' || p_string[idx + 7] != '\"') {
+			continue;
 		}
-	}
-}
-
-void ImageLoaderSVG::_convert_colors(NSVGimage *p_svg_image) {
-	for (NSVGshape *shape = p_svg_image->shapes; shape != nullptr; shape = shape->next) {
-		for (int i = 0; i < replace_colors.old_colors.size(); i++) {
-			change_nsvg_paint_color(&(shape->stroke), replace_colors.old_colors[i], replace_colors.new_colors[i]);
-			change_nsvg_paint_color(&(shape->fill), replace_colors.old_colors[i], replace_colors.new_colors[i]);
+		idx++;
+		for (int i = 0; i < 6; i++) {
+			temp_color[i] = p_string[idx + i];
 		}
-	}
-}
-
-void ImageLoaderSVG::set_convert_colors(Dictionary *p_replace_color) {
-	if (p_replace_color) {
-		Dictionary replace_color = *p_replace_color;
-		for (int i = 0; i < replace_color.keys().size(); i++) {
-			Variant o_c = replace_color.keys()[i];
-			Variant n_c = replace_color[replace_color.keys()[i]];
-			if (o_c.get_type() == Variant::COLOR && n_c.get_type() == Variant::COLOR) {
-				Color old_color = o_c;
-				Color new_color = n_c;
-				replace_colors.old_colors.push_back(old_color.to_abgr32());
-				replace_colors.new_colors.push_back(new_color.to_abgr32());
+		::Color c = ::Color::html(temp_color);
```

Comment:
Does your IDE add this scoping? I don't see why it's needed here.

Reply:
I'll fix it while doing some testing, I think the color replacement method could still be simplified, we have plenty of methods in `String` to do search and replace, doing it char by char seems pretty convoluted.

Reply:
My clang compile requires the "::" scoping.

Reply:
What clang is that? Works fine for me and on CI (macOS + iOS + Android all use clang).

We use `Color` everywhere in the codebase. What does it conflict with here?